internal/cri/server/container_status_linux.go

```diff
@@ -0,0 +1,53 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	containerstore "github.com/containerd/containerd/v2/internal/cri/store/container"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func toCRIContainerUser(ctx context.Context, container containerstore.Container) (*runtime.ContainerUser, error) {
+	if container.Container == nil {
+		return nil, nil
+	}
+
+	runtimeSpec, err := container.Container.Spec(ctx)
```

Comment:
No need to get this info directly from the container process? How setuid affects to this value?

Reply:
I wanted to extract the actual container `runtimespec.process.user.{UID, GID, AdditionalGids}`.

> No need to get this info directly from the container process?

oh really? I would be glad if you tell me how/where to get the actual process.user info??

Reply:
>> How setuid affects to this value?

> sorry. I can't get the background of this question. Could you kindly explain a bit more detail? 🙇

It seems that the user can create a situation where the container's actual uid/gid differ from the ones cached in containerd. :thinking: 

container:

```dockerfile
# syntax= docker/dockerfile:1.5

FROM gcc:14 AS dev
RUN <<EOT
cat <<EOF > test.c
#include <unistd.h>
int main(){
  setuid(123);
  sleep(10000000);
}
EOF
EOT
RUN gcc -static -o test test.c

FROM ubuntu:22.04
RUN useradd -u 123 test
COPY --from=dev /test /
USER root
ENTRYPOINT ["/test"]
```

kind example:

```console
$ kind create cluster
$ cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers:
  - name: test
    image: ghcr.io/ktock/testimg:1
EOF
$ kubectl exec -it test -c test -- cat /proc/1/status | grep Uid
Uid:	123	123	123	123
$ docker exec -it kind-control-plane ctr -n k8s.io c info --spec $(docker exec -it kind-control-plane ctr -n k8s.io c ls | grep testimg | sed -E 's/ +/ /g' | cut -f 1 -d ' ') | jq '.process.user'
{
  "uid": 0,
  "gid": 0
}
```

One possible workaround would be to directly getting uid/gid info from procfs.

Reply:
> One possible workaround would be to directly getting uid/gid info from procfs.

But it's maybe too low level and can't be applied to some low level runtimes.

Reply:
Ah, thanks for the detailed explanation. I probably got your point.

If I understood correctly, _the actual_ process identity(uid, gid, additional gids) can be dynamic theoretically if the _initially attached_ process identity can have enough privilege.

Although my wording was not appropriate, the intention of this proposed API is not to detect/inform such actual and dynamic process identity but to inform _the initially attached_ process identity which would be equivalent with OCI Runtime spec's `process.user` field.

Reply:
Thanks for clarification. Yes, it should be documented at least.

Reply:
I documented this in [baac35c](https://github.com/kubernetes/kubernetes/pull/117842/commits/baac35c1f8ae6111490fde30d340bcc223a3f1c9) in https://github.com/kubernetes/kubernetes/pull/117842. Please also see my comment in the PR: https://github.com/kubernetes/kubernetes/pull/117842#discussion_r1603043767

Reply:
TODO: add current actual user info, consider if we want this for the ps tree starting from the shim for the container or just the command task, and maybe container exec tasks?